libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.WebApi/BotMessageHandler.cs

```diff
@@ -18,15 +18,15 @@ public BotMessageHandler(BotFrameworkAdapter botFrameworkAdapter)
         {
         }
 
-        protected override async Task<InvokeResponse> ProcessMessageRequestAsync(HttpRequestMessage request, BotFrameworkAdapter botFrameworkAdapter, Func<ITurnContext, Task> botCallbackHandler, CancellationToken cancellationToken)
+        protected override async Task<InvokeResponse> ProcessMessageRequestAsync(HttpRequestMessage request, BotFrameworkAdapter botFrameworkAdapter, BotCallbackHandler botCallbackHandler, CancellationToken cancellationToken)
         {
-            var activity = await request.Content.ReadAsAsync<Activity>(BotMessageHandlerBase.BotMessageMediaTypeFormatters, cancellationToken);
+            var activity = await request.Content.ReadAsAsync<Activity>(BotMessageHandlerBase.BotMessageMediaTypeFormatters, cancellationToken).ConfigureAwait(false);
 
             var invokeResponse = await botFrameworkAdapter.ProcessActivityAsync(
                 request.Headers.Authorization?.ToString(),
                 activity,
                 botCallbackHandler,
-                cancellationToken);
+                cancellationToken).ConfigureAwait(false);
```

Comment:
So technically you want to come back to the ASP.NET context here. I would not suggest using `ConfigureAwait` in any ASP.NET component ever really because, like WinForms, they are purposefully working in a specific `SynchronizationContext`... even though it's not tied to a specific thread pool per se.